src/cli/subcommand/install.rs

```diff
@@ -133,6 +151,20 @@ impl CommandExecute for Install {
 
                 match existing_receipt {
                     Some(existing_receipt) => {
+                        if let Err(e) = existing_receipt.check_compatible() {
+                            eprintln!(
+                                "{}", 
+                                format!("\
+                                    {e}\n\
+                                    \n\
+                                    Found existing plan in `{RECEIPT_LOCATION}` which was created by a version incompatible `nix-installer`.\n\
```

Comment:
Nit: since this is used in 2 different places, maybe we could get a const or static or something so they don't drift?